Slack-Apps/week3/homework/responses/3.3-calling-translator-api.md

```diff
@@ -0,0 +1,90 @@
+
+## Country Code
+
+Now that we have the ```const country``` set to our country code, we need to find the corresponding language to our country code. To do this, we need to create a new variable that is equal to the corresponding language. To do this, we can iterate through or simply search through our langMap for the language (*remember, it has two arguments: country and language which we need to set equal to language*).
+
+Setting our Country Code: ```const [countryCode, language] = langMap[country];```
+
+## Finding Text to Translate
+
+In here, we need to pass through channel, latest, inclusive, and limit. The most important values are ```latest``` which we set to ts (*remember our const reaction?*) and ```limit``` which we set to 1 (inclusive should also be true). If you are having trouble with the layout, consider looking at how we setup the previous thread function!
+
+Previous Thread: ```await client.chat.postMessage({channel, thread_ts: ts, text: `:${reaction}: *Here is the translation of this message in ${language}:*\n ${translatedText}`,});```
+
+This ```client.conversations.history``` is super important because we need to find the text the user wants to translate! To do so, we need to set our textToTranslate equal to ```result.messages[0] (first index message!)```. We also need to add: ```const langIsSupported = langMap[country];``` to insure our language is supported.
+
+## Translation Callback Logic
+
+```javascript const [x, country] = langMap[reaction] ? [, reaction] : reaction.split('-');
+  
+    const result = await client.conversations.history({
+      channel,
+      latest: ts,
+      inclusive: true,
+      limit: 1
+    });
+  
+    const { text: textToTranslate } = result.messages[0];
+    const langIsSupported = langMap[country];
+  
+    if (langIsSupported) {
+      const [countryCode, language] = langMap[country];
+      const [translatedText, ...y] = await translateText(textToTranslate, countryCode);
+    
+      await client.chat.postMessage({
+        channel,
+        thread_ts: ts,
+        text: `:${reaction}: *Here is the translation of this message in ${language}:*\n ${translatedText}`,
+      });
+    }
+  });
+```
+Your code should look something like this besides:
+
+```const [translatedText, ...y] = await translateText(textToTranslate, countryCode);```
+
+If you noticed here, we are actually calling the ```translateText()``` function. The translateText function takes in two parameters, the text to translate and the countryCode which corresponds to the APIs recognized languages.
+
+To create the function, start off with this:
+
+```javascript 
+async function translateText(text, target) {
+    try {
+      
+    } catch (e) {
+      
+    }
+  }
+```
+
+This is a simple try and catch function. If you don't know what try and catch is, basically it processes the code in ```try()``` and if an error is found, it will return ```catch()```, it's also known as the handling of exceptions.
```

Comment:
Add a quick article in case they want to read more, and maybe explain why try catch is important especially in the industry